bot/server.go

```diff
@@ -5,119 +5,145 @@ import (
 	"fmt"
 	"net/http"
 
-	cmn "github.com/jaimeteb/chatto/common"
-	log "github.com/sirupsen/logrus"
-
 	"github.com/gorilla/mux"
+	"github.com/jaimeteb/chatto/channels"
+	"github.com/jaimeteb/chatto/channels/messages"
+	"github.com/jaimeteb/chatto/query"
+	log "github.com/sirupsen/logrus"
 )
 
-func (b Bot) restEndpointHandler(w http.ResponseWriter, r *http.Request) {
-	mess, err := b.Clients.REST.RecieveMessage(w, r)
-	if err != nil {
-		log.Error(err)
-		return
-	}
+func (b *Bot) restEndpointHandler(w http.ResponseWriter, r *http.Request) {
+	b.endpointHandler(w, r, b.Channels.REST)
+}
 
-	resp := b.Answer(mess)
+func (b *Bot) telegramEndpointHandler(w http.ResponseWriter, r *http.Request) {
+	b.endpointHandler(w, r, b.Channels.Telegram)
+}
 
-	if err := SendMessages(resp, &b.Clients.REST, mess.Sender, w); err != nil {
-		log.Error(err)
-		return
-	}
+func (b *Bot) twilioEndpointHandler(w http.ResponseWriter, r *http.Request) {
+	b.endpointHandler(w, r, b.Channels.Twilio)
 }
 
-func (b Bot) telegramEndpointHandler(w http.ResponseWriter, r *http.Request) {
-	mess, err := b.Clients.Telegram.RecieveMessage(w, r)
+func (b *Bot) slackEndpointHandler(w http.ResponseWriter, r *http.Request) {
+	b.endpointHandler(w, r, b.Channels.Slack)
+}
+
+func (b *Bot) endpointHandler(w http.ResponseWriter, r *http.Request, chnl channels.Channel) {
```

Comment:
Because we are now properly using the `Channel` interface we can generalize the endpoint handler for all our clients and reduce duplicate code.